core/http/src/uri/uri.rs

```diff
@@ -316,3 +325,200 @@ macro_rules! impl_uri_from {
+
+            test_helper("http://example.com", absolute("http", None, Some("example.com"), None, None, None));
+            test_helper("http://topleveldomain", absolute("http", None, Some("topleveldomain"), None, None, None));
+            test_helper("http://sub.example.com", absolute("http", None, Some("sub.example.com"), None, None, None));
+            test_helper("http://user@example.com", absolute("http", Some("user"), Some("example.com"), None, None, None));
+            test_helper("http://user:pass@example.com", absolute("http", Some("user:pass"), Some("example.com"), None, None, None));
+            test_helper("http://example.com:80", absolute("http", None, Some("example.com"), Some(80), None, None));
+            test_helper("http://example.com:123", absolute("http", None, Some("example.com"), Some(123), None, None));
+            test_helper("http://example.com:65333", absolute("http", None, Some("example.com"), Some(65333), None, None));
+            test_helper("http://user@example.com:80", absolute("http", Some("user"), Some("example.com"), Some(80), None, None));
+            test_helper("http://user:pass@example.com:80", absolute("http", Some("user:pass"), Some("example.com"), Some(80), None, None));
+
+            test_helper("http:///path", absolute("http", None, Some(""), None, Some("/path"), None));
+            test_helper("http:///path/", absolute("http", None, Some(""), None, Some("/path/"), None));
+            test_helper("http:///path/with/segments", absolute("http", None, Some(""), None, Some("/path/with/segments"), None));
+            test_helper("http:///path/with/segments/", absolute("http", None, Some(""), None, Some("/path/with/segments/"), None));
+            test_helper("http:///index.html", absolute("http", None, Some(""), None, Some("/index.html"), None));
+            test_helper("http:///path?query", absolute("http", None, Some(""), None, Some("/path"), Some("query")));
+            test_helper("http:///path?query=value", absolute("http", None, Some(""), None, Some("/path"), Some("query=value")));
+            test_helper("http:///index.html?query", absolute("http", None, Some(""), None, Some("/index.html"), Some("query")));
+            test_helper("http:///index.html?query=value", absolute("http", None, Some(""), None, Some("/index.html"), Some("query=value")));
+            test_helper("http:///index.html?query", absolute("http", None, Some(""), None, Some("/index.html"), Some("query")));
+            test_helper("http:///path/with/segments/?query=value", absolute("http", None, Some(""), None, Some("/path/with/segments/"), Some("query=value")));
+        }
+
+        #[test]
+        fn serde_asterisk_form() {
+            test_helper("*", Uri::Asterisk);
+        }
+
+        // Commented out tests currently fail
+        #[test]
+        fn serde_invalid_uri() {
+            assert_de_tokens_error::<Uri<'static>>(&[Token::Str("@")], "unexpected EOF: expected some token at index 0");
```

Comment:
I don't think we want to match on the exact error message. We already have tests for this in `uri.rs`. Simply check that parsing failed.

Reply:
I would have implemented it like that, however serde_test does require it, and I don't currently see a way around it. Please let me know if there is something I missed, but I don't think there is a way around this. I will be adding a comment to explain why and how the errors where created, but I don't think there is a way to avoid specifying them.